src/nvim/message.c

```diff
@@ -134,24 +134,23 @@ int verb_msg(char_u *s)
   return n;
 }
 
-int msg_attr(const char *s, const int attr) FUNC_ATTR_NONNULL_ARG(1)
+int msg_attr(const char *s, const int attr)
+  FUNC_ATTR_NONNULL_ARG(1)
 {
   return msg_attr_keep((char_u *)s, attr, false);
 }
 
-int msg_echo_attr(const char *s, const int attr) FUNC_ATTR_NONNULL_ARG(1)
+int msg_echo_attr(const char *s, const int attr)
+  FUNC_ATTR_NONNULL_ALL
 {
   return msg_echo_attr_keep(s, attr, false);
 }
 
-int
-msg_echo_attr_keep(const char *s, const int attr, int keep)
+bool msg_attr_skip(const char *s, const int attr, int *const call)
```

Comment:
@ZyX-I I move the common code into this function that return when the message doesn't need to be shown (I will add comments, I forgot them). Is the function ok for you? 
Also, i didn't put the `add_msg_hist` here because (if I understood well the code) it is already done in `msg_outtrans_len_attr` and `msg_puts_attr_len`. I guess it doesn't need to be done multiple time?